### PR TITLE
[DJ][FEAT-19]: Integrate accommodation api with frontend

### PR DESCRIPTION
**Pull request by purple-mountain**
# Summary
- [x] Use Axios and react query to fetch accommodations from the backend.

Right now there is no category endpoint for fetching categories so clicking on category won't filter accommodations properly because categories on the frontend need to be exactly the same as on the backend.

But filtering by category works if both are the same and this can be tested by manually changing parameter in the URL with category ID from the database.

### CODE REVIEW

File: src/modules/home/components/category/category-list.tsx [hunk 1]
```diff
@@ -2,25 +2,26 @@ import Box from '@mui/material/Box';
...
   const [showLeftArrow, setShowLeftArrow] = useState<boolean>(false);
   const [showRightArrow, setShowRightArrow] = useState<boolean>(false);
 
-  const handleCategoryClick = useCallback(
-    (categoryName: string) => {
-      navigate(`/?category=${encodeURIComponent(categoryName.toLowerCase())}`);
-    },
-    [navigate]
-  );
+  const handleCategoryClick = (category: string) => {
```
Comment (MdRasulov) [inline src/modules/home/components/category/category-list.tsx:18]: this function can be simplified to this:
```js
const handleCategoryClick = (category: string) => 
  setSelectedCategory(category === selectedCategory ? '' : category);
```

File: src/modules/home/components/property/index.tsx [hunk 2]
```diff
@@ -1,8 +1,29 @@
 import Box from '@mui/material/Box';
-import { ApartmentCard } from '#/modules/home/components/property/propertyCard';
-import data from './dummy.json';
+import { ApartmentCard } from '#/modules/home/components/property/property-card';
+import { useProperties } from '../../api/get-properties';
+import Typography from '@mui/material/Typography';
+
+export function PropertyList({ selectedCategory }: { selectedCategory: string }) {
+  const propertiesQuery = useProperties({ page: 1, categoryId: selectedCategory });
```
Comment (MdRasulov) [inline src/modules/home/components/property/index.tsx:7]: how about using Destructuring to get data and isLoading properties at once?

File: src/modules/home/components/property/index.tsx [hunk 3]
```diff
@@ -18,9 +39,9 @@ export function PropertyList() {
         py: { xs: 2, sm: 3, md: 4 },
       }}
     >
-      {data.map((item) => (
-        <ApartmentCard key={item.id} {...item} />
-      ))}
+      {propertiesData.map((item) => {
+        return <ApartmentCard key={item.id} {...item} />;
```
Comment (MdRasulov) [inline src/modules/home/components/property/index.tsx:43]: Any reasons changing from `() => ()` to `() => {return ...}`?

**Review by MdRasulov — APPROVED** [review]
Approved with some comments!

File: src/modules/home/components/property/index.tsx (hunk 3 above)
Comment (purple-mountain) [reply src/modules/home/components/property/index.tsx:43]: Initially, I was thinking of doing some if checks here but then it became unnecessary. I will turn it back.

**Pull request merged**